plugin/path_impersonated_account_secrets.go

```diff
@@ -0,0 +1,87 @@
+package gcpsecrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+)
+
+func pathImpersonatedAccountSecretAccessToken(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s/token", impersonatedAccountPathPrefix, framework.GenericNameRegex("name")),
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Description: "Required. Name of the impersonated account.",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation:   &framework.PathOperation{Callback: b.pathImpersonatedAccountAccessToken},
+			logical.UpdateOperation: &framework.PathOperation{Callback: b.pathImpersonatedAccountAccessToken},
+		},
+		HelpSynopsis:    pathTokenHelpSyn,
+		HelpDescription: pathTokenHelpDesc,
+	}
+}
+
+func (b *backend) pathImpersonatedAccountAccessToken(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	acctName := d.Get("name").(string)
+
+	acct, err := b.getImpersonatedAccount(acctName, ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if acct == nil {
+		return logical.ErrorResponse("impersonated account %q does not exists", acctName), nil
+	}
+
+	creds, err := b.credentials(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := getConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	acctTtl := time.Duration(acct.Ttl) * time.Second
+	if acctTtl > config.MaxTTL {
+		b.Logger().Debug("using backend max ttl which is less than impersonated account ttl for token",
```

Comment:
I think I agree with adding warnings with the response, however I don't think it is necessary to keep these as debug logs in that case since the debug logs are server logs and the warnings will end up in the audit logs.